app/src/main/java/org/sil/gatherwords/DatabaseAccess.java

```diff
@@ -0,0 +1,116 @@
+package org.sil.gatherwords;
+
+import android.os.AsyncTask;
+
+import org.sil.gatherwords.room.AppDatabase;
+import org.sil.gatherwords.room.Session;
+import org.sil.gatherwords.room.SessionDao;
+import org.sil.gatherwords.room.Word;
+import org.sil.gatherwords.room.WordDao;
+
+import java.util.List;
+
+/**
+ * Accesses the database in a worker thread.
+ * Prevents blocking the main thread
+ */
+
+public class DatabaseAccess extends AsyncTask<String, Void, Void> {
```

Comment:
It is general practice for async tasks to have a single purpose.  As such, smaller ones (like the insert done here) are often implemented as private nested classes.  For example, paste:

```java
    private static class InsertSessionsTask extends AsyncTask<Session, Void, Void> {

        private SessionDao sDAO;

        public InsertSessionsTask(AppDatabase db) {
            sDAO = db.sessionDao();
        }

        @Override
        protected Void doInBackground(Session... sessions) {
            sDAO.insertSession(sessions);
            return null;
        }
    }
```

into `SessionActivity` after `save_settings_fab_pressed()`, and use it by `new InsertSessionsTask(appDatabase).execute(session);`.

Reply:
Would you like to add a new class to every activity that needs to read or write to the database? I am opposed to that much extra work.